[sciphi-123] iP

sciphi-123 opened the pull request:
# Phi

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Phi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/sciphi-123/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉  

And it is **FREE!**  

## Features:

- [x] Managing tasks  
- [ ] Managing deadlines (coming soon)  
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sciphi-123 committed:
Rename, Greet, Exit

sciphi-123 committed:
Echo

sciphi-123 committed:
Echo

sciphi-123 committed:
Add, List

sciphi-123 committed:
Mark as Done

sciphi-123 committed:
ToDo, Event, Deadline

sciphi-123 committed:
A-TextUiTesting

sciphi-123 committed:
Handle errors

sciphi-123 committed:
Add support for deleting tasks

sciphi-123 committed:
Add functionality for save and load

sciphi-123 committed:
Merge conflicts with branch-Level-7

sciphi-123 committed:
Update functionality for save and load

sciphi-123 committed:
Handle datetime

sciphi-123 committed:
Update Phi.java

sciphi-123 committed:
Edit minor details

sciphi-123 committed:
Remove echo

sciphi-123 committed:
Remove echo and exit

sciphi-123 committed:
Merge branch 'branch-Level-8'

sciphi-123 committed:
Ensure echo and exit are restored

sciphi-123 committed:
Restore echo and exit

sciphi-123 committed:
Merge branch 'branch-Level-8'

sciphi-123 committed:
Reorganise code to make more OOP

sciphi-123 committed:
Add Java files into phi package

sciphi-123 committed:
Update runtest.bat

sciphi-123 committed:
Merge branch 'add-gradle-support'

sciphi-123 committed:
Add JUnit tests

sciphi-123 committed:
Update JUnit tests

sciphi-123 committed:
Create a JAR file

sciphi-123 committed:
Add JavaDoc comments

sciphi-123 committed:
Update phi.txt

sciphi-123 committed:
Merge branch 'branch-A-JavaDoc'

sciphi-123 committed:
Make some minor adjustments

sciphi-123 committed:
Update to comply with coding standard

sciphi-123 committed:
Merge branch 'branch-A-CodingStandard'

sciphi-123 committed:
Implement find method

sciphi-123 committed:
Merge branch 'branch-Level-9'

sciphi-123 committed:
Add some Javadoc comments

namitdeb739 reviewed (commented):
LGTM!

dragonesejosh commented on src/main/java/phi/Phi.java:
```diff
@@ -0,0 +1,119 @@
+                String line = ui.readCommand();
+                isRunning = this.checkInput(line);
+                storage.saveTasks(tasks);
+            } catch (PhiException e) {
+                System.out.println("Error: " + e.getMessage());
+                System.out.println();
+            } catch (NumberFormatException e) {
+                System.out.println("Give a number!");
+                System.out.println();
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("The number entered is not within range!");
+                System.out.println();
+            } catch (DateTimeParseException e) {
+                System.out.println("Please enter valid date(s) in the format yyyy-mm-dd.");
+                System.out.println();
+            }
```
I like that you specifically catch different exceptions and handle each one individually.

dragonesejosh commented on text-ui-test/data/phi.txt:
```diff
@@ -0,0 +1,4 @@
+[T][ ] read book
+[E][ ] project meeting (from: Feb 2 2019 to: Feb 3 2019)
+[D][ ] finish project (by: Jan 10 2020)
+[E][ ] team outing (from: Jan 1 2021 to: Jan 2 2021)
```
You might want to add `phi.txt` to the `.gitignore` file, since it's a local file that changes when you run the program.

dragonesejosh commented on src/main/java/phi/Storage.java:
```diff
@@ -0,0 +1,117 @@
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
```
You might want to replace all instances of `System.out.println` outside of `Ui` with calls to a `Ui` instance, especially for when you need to move your output into a GUI in the Week 4 task.
e.g. `ui.printError("Error" + ...);`

dragonesejosh commented on src/main/java/phi/Storage.java:
```diff
@@ -0,0 +1,117 @@
+     */
+    public void saveTasks(TaskList tasks) {
+        try {
+            File dir = new File("./data/");
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
```
Is there a reason why you're creating a fixed directory instead of the specified `textFilePath`?

dragonesejosh reviewed (commented):
Looks good, just some minor changes might be needed.

sciphi-123 committed:
Add some GUI features

sciphi-123 committed:
Merge branch 'branch-Level-10'

sciphi-123 committed:
Improve GUI features

sciphi-123 committed:
Add Javadoc comments

sciphi-123 committed:
Add checkstyle edits

sciphi-123 committed:
Add more checkstyle edits

sciphi-123 committed:
Add assertions for important assumptions

There previously was no assertions to
ensure that specific lines run appropriately.

This could make code hard to debug in the event
of an error.

Assertions are added in getResponse() and
checkInput() methods of Phi class, to check for:

getResponse():
- Input is not null

checkInput():
- Check that the size of the list of tasks change as expected

Such assertions would be helpful to ensure
that the code could be debugged properly should
it crash unexpectedly.

sciphi-123 committed:
Improve code quality with style adjustments

- Updated several Javadoc descriptions
so they more accurately relate to what is
presented in the code

- Ran checkstyle on the code and resolved
as many errors as possible

These changes would improve the overall
quality of the code for the developer.

sciphi-123 committed:
Merge pull request #2 from sciphi-123/branch-A-Assertions

Add assertions for important assumptions

sciphi-123 committed:
Merge branch 'master' into branch-A-CodeQuality

sciphi-123 committed:
Merge pull request #3 from sciphi-123/branch-A-CodeQuality

Improve code quality with style adjustments

sciphi-123 committed:
Add sort tasks functionality

sciphi-123 committed:
Merge branch 'master' of https://github.com/sciphi-123/ip

sciphi-123 committed:
Add minor edits for better user experience

sciphi-123 committed:
Make adjustments based on code quality feedback

sciphi-123 committed:
Add user guide and some code edits

sciphi-123 committed:
Make further edits to user guide

sciphi-123 committed:
Remove some extraneous comments

sciphi-123 committed:
Add style edits

sciphi-123 committed:
Make minor code edits

sciphi-123 committed:
Make style edit

sciphi-123 committed:
Account for whitespace in command line

sciphi-123 committed:
Make minor edits
